ethers-core/src/types/block.rs

```diff
@@ -624,15 +639,22 @@ impl FromStr for BlockNumber {
     type Err = String;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let block = match s {
-            "latest" => Self::Latest,
-            "finalized" => Self::Finalized,
-            "safe" => Self::Safe,
-            "earliest" => Self::Earliest,
-            "pending" => Self::Pending,
-            n => BlockNumber::Number(n.parse::<U64>().map_err(|err| err.to_string())?),
-        };
-        Ok(block)
+        match s {
+            "latest" => Ok(Self::Latest),
+            "finalized" => Ok(Self::Finalized),
+            "safe" => Ok(Self::Safe),
+            "earliest" => Ok(Self::Earliest),
+            "pending" => Ok(Self::Pending),
+            n => {
+                if let Ok(n) = n.parse::<U64>() {
+                    Ok(Self::Number(n))
+                } else if let Ok(n) = n.parse::<u64>() {
+                    Ok(Self::Number(n.into()))
```

Comment:
This is no longer compliant with the rpc spec for block number, but I guess this is fine?

Reply:
Fine w/ me. More flexibility, and people create `BlockNumber` themselves, vs always parsing it out from the hex returned from RPC